linker/shared/src/main/scala/org/scalajs/linker/standard/IOThrottler.scala

```diff
@@ -15,18 +15,17 @@ package org.scalajs.linker.standard
 import scala.annotation.tailrec
 import scala.concurrent._
 
-import java.util.concurrent.ConcurrentLinkedQueue
-import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.{ConcurrentLinkedQueue, Semaphore}
 
 private[linker] final class IOThrottler(totalSlots: Int) {
   /* This is basically a java.util.concurrent.Semaphore, but it is not
    * implemented in the javalib.
    */
-  private val slots = new AtomicInteger(totalSlots)
+  private val semaphore = new Semaphore(totalSlots)
```

Comment:
I'd keep calling this `slots` (call it what it represents, not what it is. notably, it wasn't called `atomicInteger` before).

Reply:
:+1: